### ISSUE TITLE
Autoclosing ruby blocks

### ISSUE THREAD

**Issue by Flink** [spf13/spf13-vim#280]
Hi!

I don’t know if this is really an issue, but still it is one for me. I’m currently using Janus and I’m thinking to switch to spf13.
When editing a ruby source file, I tried to create a new method. When hitting enter after `def method_name` I was expecting to have the block automatically completed (with `end`) as Janus does.
So far I only found documentation of vim-ruby saying that Shift+Enter should do the trick, but here it gives me an error (related to completion it seems).

Anyone knows how to enable this behavior?

**Comment by spf13** [comment]
## I'll look into it.

Steve Francia
http://stevefrancia.com
http://spf13.com
http://twitter.com/spf13

**Comment by Flink** [comment]
Thank you! :)

**Comment by hsingh23** [comment]
You are looking for Bundle 'tpope/vim-endwise'

Make your Ruby bundle look like this
    " Ruby
        if count(g:harsh_bundle_groups, 'ruby')
            Bundle 'tpope/vim-rails'
            Bundle 'gmarik/vundle'
            Bundle 'tpope/vim-cucumber'
            Bundle 'tpope/vim-endwise'
            Bundle 'vim-ruby/vim-ruby'
            let g:rubycomplete_buffer_loading = 1
            "let g:rubycomplete_classes_in_global = 1
            "let g:rubycomplete_rails = 1
        endif

**Comment by chrisnicola** [comment]
The OmniComplete settings creats some havoc with this. You end up inserting `pubvisible()...` after hitting enter unless you change the <CR> mapping for it. Which is fine but you do have to comment out this line in .vimrc to make this work:

`"inoremap <expr> <CR>       pumvisible() ? "\<C-y>" : "\<CR>"`

**Comment by spf13** [comment]
Does someone want to send me a PR for these changes?

Best,
Steve

-- 
Steve Francia
http://stevefrancia.com
http://spf13.com
http://twitter.com/spf13

**Comment by chrisnicola** [comment]
Yeah I'll put something together. Do you want to remove the <CR> mapping altogether (I'm not certain what it actually does since tab is what completes things for me, the <CR> mapping's behaviour is awkward). The alternative is I can add a variable for not mapping CR for OmniComplete that can be used instead.

**Comment by spf13** [comment]
Let's add a variable, but have the default set to false ( not include it ).

On Wednesday, June 18, 2014, Chris Nicola notifications@github.com wrote:

> Yeah I'll put something together. Do you want to remove the mapping
> altogether (I'm not certain what it actually does since tab is what
> completes things for me, the mapping's behaviour is awkward). The
> alternative is I can add a variable for not mapping CR for OmniComplete
> that can be used instead.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/spf13/spf13-vim/issues/280#issuecomment-46499346.

## 

## 

Steve Francia
http://stevefrancia.com
http://spf13.com
http://twitter.com/spf13

**Comment by bronzehedwick** [comment]
@Flink, have the fixes here resolved the issue? If so please close :)
